backend/internal/branding/resolve/service.go

```diff
@@ -0,0 +1,119 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package brandingresolve
+
+import (
+	"strings"
+
+	"github.com/asgardeo/thunder/internal/application"
+	"github.com/asgardeo/thunder/internal/branding/common"
+	brandingmgt "github.com/asgardeo/thunder/internal/branding/mgt"
+	"github.com/asgardeo/thunder/internal/system/error/serviceerror"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+const (
+	serviceLogger  = "BrandingResolveService"
+	resolveTypeAPP = "APP"
+)
+
+// BrandingResolveServiceInterface defines the interface for the branding resolve service.
+type BrandingResolveServiceInterface interface {
+	ResolveBranding(resolveType, id string) (*common.BrandingResponse, *serviceerror.ServiceError)
+}
+
+// brandingResolveService is the default implementation of the BrandingResolveServiceInterface.
+type brandingResolveService struct {
+	brandingMgtService brandingmgt.BrandingMgtServiceInterface
+	applicationService application.ApplicationServiceInterface
+	logger             *log.Logger
+}
+
+// newBrandingResolveService creates a new instance of BrandingResolveService with injected dependencies.
+func newBrandingResolveService(
+	brandingMgtService brandingmgt.BrandingMgtServiceInterface,
+	applicationService application.ApplicationServiceInterface,
+) BrandingResolveServiceInterface {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, serviceLogger))
+	return &brandingResolveService{
+		brandingMgtService: brandingMgtService,
+		applicationService: applicationService,
+		logger:             logger,
+	}
+}
+
+// ResolveBranding resolves a branding configuration by type and ID.
+// TODO: Add support for OU type and fallback logic.
+func (brs *brandingResolveService) ResolveBranding(
+	resolveType, id string,
+) (*common.BrandingResponse, *serviceerror.ServiceError) {
+	resolveTypeUpper := strings.ToUpper(resolveType)
+
+	if resolveTypeUpper == "" {
+		return nil, &common.ErrorInvalidResolveType
+	}
+
+	if id == "" {
+		return nil, &common.ErrorMissingResolveID
+	}
+
+	// Currently only APP type is supported
+	if resolveTypeUpper != resolveTypeAPP {
+		return nil, &common.ErrorUnsupportedResolveType
+	}
+
+	// Get the application by ID
+	if brs.applicationService == nil {
+		brs.logger.Error("Application service is not available")
+		return nil, &serviceerror.InternalServerError
+	}
+
+	app, svcErr := brs.applicationService.GetApplication(id)
+	if svcErr != nil {
+		// Convert application service errors to branding resolve errors
+		if svcErr.Code == "APP-1001" || svcErr.Code == "APP-1004" {
```

Comment:
Hard coding error codes is not a good practice. We should import error constants from application package and retrieve the code